java/src/jni/h5util.c

```diff
@@ -813,7 +813,7 @@ h5str_sprintf(JNIEnv *env, h5str_t *out_str, hid_t container, hid_t tid, void *i
                     if (NULL == (this_str = (char *)HDmalloc(this_len)))
                         H5_OUT_OF_MEMORY_ERROR(ENVONLY, "h5str_sprintf: failed to allocate string buffer");
 
-                    if (HDsnprintf(this_str, "%g", this_len, tmp_double) < 0)
+                    if (HDsnprintf(this_str, this_len, "%g", tmp_double) < 0)
```

Comment:
This code likely started out in the tools lib, might want to dbl-chk the str_sprintf function in there.